template/build.sh

```diff
@@ -0,0 +1,39 @@
+#!/usr/bin/env bash
+repoFolder="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+cd $repoFolder
+
+koreBuildZip="https://github.com/aspnet/KoreBuild/archive/dev.zip"
+if [ ! -z $KOREBUILD_ZIP ]; then
+    koreBuildZip=$KOREBUILD_ZIP
+fi
+
+buildFolder=".build"
+buildFile="$buildFolder/KoreBuild.sh"
+
+if test ! -d $buildFolder; then
+    echo "Downloading KoreBuild"
+    
+    tempFolder="/tmp/KoreBuild-$(uuidgen)"
+    
+    if test ! -d $tempFolder; then
+        rm -rf $tempFolder  
+    fi
+    
+    mkdir $tempFolder
+    
+    localZipFile="$tempFolder/korebuild.zip"
+    
+    wget -O $localZipFile $koreBuildZip 2>/dev/null || curl -o $localZipFile --location $koreBuildZip /dev/null
+    unzip -q -d $tempFolder $localZipFile
+  
+    mkdir $buildFolder
+    cp -r $tempFolder/**/build/** $buildFolder
+    
+    chmod +x $buildFile
+    
+    if test ! -d $tempFolder; then
```

Comment:
`test -d $tempFolder;` since this block should execute if the folder _does_ exist.